src/object/obj_class.c

```diff
@@ -482,3 +496,62 @@ obj_ec_codec_get(daos_oclass_id_t oc_id)
 
 	return NULL;
 }
+
+
+/* Encode full stripe from SGL */
+int
+obj_encode_full_stripe(daos_obj_id_t oid, daos_sg_list_t *sgl, uint32_t *j,
+		       size_t *k, struct obj_ec_parity *parity, int p_idx)
+{
+	struct obj_ec_codec 		*codec =
+					 obj_ec_codec_get(daos_obj_id2class(oid));
+ 	struct daos_oclass_attr 	*oca =
+					 daos_oclass_attr_find(oid);
+	unsigned int    		 clen = oca->u.ec.e_len;
+	unsigned int    		 dc = oca->u.ec.e_k;
+	unsigned int			 p = oca->u.ec.e_p;
+	unsigned char 			*data[dc];
+	unsigned char 			*ldata[dc];
+	int				 i, lcnt = 0;
+	int				 rc = 0;
+	
+	for (i = 0; i < dc; i++) 
+		if (sgl->sg_iovs[*j].iov_len - *k >= clen) {
+			unsigned char* from =
+				(unsigned char*)sgl->sg_iovs[*j].iov_buf;
+			data[i] = &from[*k];
+			*k += clen;
+			if (*k == sgl->sg_iovs[*j].iov_len) {
+				*k = 0;
+				(*j)++;
+			}
+		} else {
+			int cp_cnt = 0;
+			D_ALLOC_ARRAY(ldata[lcnt], clen);
+			if (ldata[lcnt] == NULL)
+				D_GOTO(out, rc = -DER_NOMEM);
+			while (cp_cnt < clen) {
+				int cp_amt = sgl->sg_iovs[*j].iov_len-*k <
+					clen - cp_cnt ?
+					sgl->sg_iovs[*j].iov_len-*k :
+					clen - cp_cnt;
+				unsigned char* from = sgl->sg_iovs[*j].iov_buf;
+				memcpy(&ldata[lcnt][cp_cnt], &from[*k], cp_amt);
+				if (sgl->sg_iovs[*j].iov_len-*k < clen - cp_cnt) {
```

Comment:
(style)  line over 80 characters
(style)  suspect code indent for conditional statements (32, 41)